### ISSUE TITLE
Support fo runit init system

### ISSUE THREAD

**Issue by argenkiwi** [rvaiya/keyd#760]
Would it be possible to add support for the runit init system? 

I noticed that building from source and installing keyd applies the necessary changes to launch the keyd service under systemd. However, it seems some extra work is needed to make it work with distros which use the runit init system.

I use Void Linux and I love the fact that keyd is available via it's package manager. However, the latest release is a bit outdated and I don't get the benefits of the latest bug fixes. I am particularly interested in the fix that allows `swap` to work with `overloadt` which is important for the [layout](https://github.com/argenkiwi/kenkyo) I have been working on.

Alternatively, a fresh release would do the trick (I noticed it was the birthday of the current one 10 days ago). Thanks!

**Comment by SweaterFish** [comment]
I use keyd with runit in antiX and have made my own runit script. It's pretty easy since keyd is just a simple daemon.

In fact, if Void supports /etc/rc.local, that's probably the best way to start keyd

Otherwise, here's my runit script if you want to use it as a starting point. The only oddity I encountered on my system was that without the short `sleep` before starting the daemon, keyd would start without errors, but mappings didn't work. I assume there's something related to the default keymapping needs to finish. Someone else might be able to explain what exactly and then a proper runit script that checks and waits for that something can be made.

My /etc/sv/runit/run script:
`#!/usr/bin/env /lib/runit/invoke-run
set -e

NAME="keyd"
DAEMON=/usr/bin/keyd

# Exit service if DAEMON is not installed
if [ ! -x $DAEMON ]; then
	exit 161
fi

exec 2>&1

sleep 0.1

exec $DAEMON
`

**Comment by argenkiwi** [comment]
Thanks for sharing @SweaterFish. I managed to solve this in my instance of Void Linux by adding a service manually. I copied the `run` script that is added when installing `keyd` from XBPS and adjused the call to `keyd` to be by its absolute path. There is a comment in there referring to the exact issue you mention and explains what the problem is (I am away from my machine at the moment and cannot recall the details).

It might not be possible to support runit by default if the location of the services folder is not consistent in all runit-based distributions. I didn't find another issue talking about it so I thought I'd raise one.
